services/src/floodlight-modules/src/main/java/org/openkilda/floodlight/switchmanager/SwitchManager.java

```diff
@@ -1474,7 +1480,53 @@ public void sendPortUpEvents(final IOFSwitch sw) throws SwitchOperationException
                 }
             }
         }
+    }
+    
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public PortConfigurationResponse configurePort(final PortConfigurationRequest request) 
+            throws SwitchOperationException {
+        DatapathId dpId = DatapathId.of(request.getSwitchId());
+        IOFSwitch sw = lookupSwitch(dpId);
+        
+        OFPortDesc ofPortDesc = getPort(sw, request.getPortNumber());
+        
+        boolean result = false;
+        if (request.getStatus() != null) {
+            result = updatePortStatus(sw, ofPortDesc, request.getStatus());
+        }
 
+        // TODO Update port speed code
+        return new PortConfigurationResponse(request.getSwitchId(), request.getPortNumber(), result);
     }
+    
+    private boolean updatePortStatus(final IOFSwitch sw, final OFPortDesc ofPortDesc, final PortStatus portStatus) {
+        Set<OFPortConfig> configs = portStatus == PortStatus.UP ? configs = ImmutableSet.of() : 
```

Comment:
To change port configuration, I have to send OFPortMod message to switch.
OFPortMod requires port no, hw address, configuration,  mask. All Fields are mandatory. 

In this PR only port up and down are requested, no change in HW address. If we don't set HW address it will pic default mac address and write on the switch. To avoid this we pass the existing mac address in the request.
If there is a better way please explain.